Source/WebInspectorUI/UserInterface/Models/ScriptInstrument.js

```diff
@@ -34,20 +34,30 @@ WI.ScriptInstrument = class ScriptInstrument extends WI.Instrument
 
     startInstrumentation(initiatedByBackend)
     {
-        let target = WI.assumingMainTarget();
-
         // FIXME: Make this some UI visible option.
         const includeSamples = true;
 
-        if (!initiatedByBackend)
-            target.ScriptProfilerAgent.startTracking(includeSamples);
+        if (!initiatedByBackend) {
+            for (let target of WI.targets) {
+                if (target.type === WI.TargetType.Worker && !WI.settings.experimentalEnableWorkerTimelineRecording.value)
```

Comment:
Idea: we may want to move towards target iteration lambdas once there are multiple target types to iterate through. (It will also allow pushing the experimental setting check down.)

i.e., `forEachWorkerTarget(...)`